examples/basic.zig

```diff
@@ -0,0 +1,52 @@
+const std = @import("std");
+const zig_pdk = @import("extism-zig-pdk");
+const pdk = zig_pdk.Plugin;
+const http = zig_pdk.http;
+
+pub fn main() void {}
+const allocator = std.heap.wasm_allocator;
+
+export fn count_vowels() i32 {
+    const plugin = pdk.init(allocator);
+    const input = plugin.getInput() catch unreachable;
+    defer allocator.free(input);
+    var count: i32 = 0;
+    for (input) |char| {
+        switch (char) {
+            'A', 'I', 'E', 'O', 'U', 'a', 'e', 'i', 'o', 'u' => count += 1,
+            else => {},
+        }
+    }
+    var var_a_optional = plugin.getVar("a") catch unreachable;
+    if (var_a_optional == null) {
+        plugin.setVar("a", "this is var a");
+    } else {
+        allocator.free(var_a_optional.?);
+    }
+    const var_a = plugin.getVar("a") catch unreachable orelse "";
+    defer allocator.free(var_a);
+
+    const thing = plugin.getConfig("thing") catch unreachable orelse "<unset by host>";
+
+    const output = std.fmt.allocPrint(allocator, "{{\"count\": {d}, \"config\": \"{s}\", \"a\": \"{s}\"}}", .{ count, thing, var_a }) catch unreachable;
```

Comment:
Out of curiosity, does Zig have a json encoder or decoder implementation that can be used from Wasm?

Reply:
Yes it does in std.json namespace, I just wrote it in plaintext for the sake of simplicity.